src/sql/workbench/contrib/query/test/browser/queryInputFactory.test.ts

```diff
@@ -141,6 +141,217 @@ suite('Query Input Factory', () => {
 		assert(connectionManagementService.numberConnects === 0, 'Convert input should not have been called connect when no global connections exist');
 	});
 
+	test('sync query editor input can only be connected once when global connection exists', () => {
+		const instantiationService = workbenchInstantiationService();
+		const editorService = new MockEditorService();
+		const connectionManagementService = new MockConnectionManagementService();
+		instantiationService.stub(IObjectExplorerService, new MockObjectExplorerService());
+		instantiationService.stub(IConnectionManagementService, connectionManagementService);
+		instantiationService.stub(IEditorService, editorService);
+		const queryEditorLanguageAssociation = instantiationService.createInstance(QueryEditorLanguageAssociation);
+		const input1 = createFileInput(URI.file('/test/file.sql'), undefined, undefined, undefined);
+		const input2 = createFileInput(URI.file('/test/file.sql'), undefined, undefined, undefined);
+		queryEditorLanguageAssociation.syncConvertInput(input1);
+		queryEditorLanguageAssociation.syncConvertInput(input2);
+		let connProfile = connectionManagementService.getConnectionProfile('file:///test/file.sql');
+		assert(connProfile !== undefined, 'connection profile should not be undefined');
+		assert(connProfile.azureResourceId === 'file:///test/file.sql', 'azureResourceId should not be undefined');
```

Comment:
Why are you testing the azureResourceId? And why is it being set to the file URI? That doesn't seem right - the resource ID should correspond to a value from this enum : https://github.com/microsoft/azuredatastudio/blob/main/src/sql/azdata.d.ts#L2277

Reply:
Changing it to return actual profile